lxd/storage_volumes.go

```diff
@@ -1040,43 +1040,65 @@ func storagePoolVolumeTypeDelete(d *Daemon, r *http.Request, volumeTypeName stri
 		}
 	}
 
-	s, err := storagePoolVolumeInit(d.State(), "default", poolName, volumeName, volumeType)
-	if err != nil {
-		return response.NotFound(err)
-	}
-
-	switch volumeType {
-	case storagePoolVolumeTypeCustom:
-		var snapshots []string
+	// Check if we can load new storage layer for pool driver type.
+	pool, err := storagePools.GetPoolByName(d.State(), poolName)
+	if err != storageDrivers.ErrUnknownDriver {
+		if err != nil {
+			return response.SmartError(err)
+		}
 
-		// Delete storage volume snapshots
-		snapshots, err = d.cluster.StoragePoolVolumeSnapshotsGetType(volumeName, volumeType, poolID)
+		switch volumeType {
+		case storagePoolVolumeTypeCustom:
+			err = pool.DeleteCustomVolume(volumeName, nil)
+		case storagePoolVolumeTypeImage:
+			err = pool.DeleteImage(volumeName, nil)
+		default:
+			return response.BadRequest(fmt.Errorf(`Storage volumes of type "%s" `+
```

Comment:
I tend to prefer we don't wrap such error strings, makes it much easier to grep.

Reply:
Have fixed this.